packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -994,45 +1003,75 @@ function simplifyIndexed<T>(
 }
 
 function simplifyUnion<T>(typeDef: UnionType, context: SimplifyContext<T>): SimplifiedMcdocType {
-	const members: SimplifiedMcdocTypeNoUnion[] = []
 	const filterCanonical = context.ctx.requireCanonical
 		&& typeDef.members.some(m => m.attributes?.some(a => a.name === 'canonical'))
-	for (const member of typeDef.members) {
-		if (filterCanonical && !member.attributes?.some(a => a.name === 'canonical')) {
-			continue
-		}
-		const simplified = simplify(member, context)
-		let keep = true
-		handleAttributes(member.attributes, context.ctx, (handler, config) => {
-			if (!keep || !handler.filterElement) return
-			if (!handler.filterElement(config, context.ctx)) {
-				keep = false
+
+	const validMembers = typeDef.members
+		.filter(member => {
+			if (filterCanonical && !member.attributes?.some(a => a.name === 'canonical')) {
+				return false
 			}
+			let keep = true
+			handleAttributes(member.attributes, context.ctx, (handler, config) => {
+				if (!keep || !handler.filterElement) return
+				if (!handler.filterElement(config, context.ctx)) {
+					keep = false
+				}
+			})
+			return keep
 		})
-		if (!keep) continue
 
-		if (simplified.kind === 'union') {
-			members.push(...simplified.members)
-		} else {
-			members.push(simplified)
+	const members = validMembers
+		.flatMap(member => {
+			const simplified = simplify(member, { ...context, spreadingUnion: true })
+			if (simplified.kind === 'union') {
+				return simplified.members
+			}
+			return [simplified]
+		})
+
+	if (context.literalStructFields && !context.spreadingUnion) {
```

Comment:
Is it intended that complex fields are not handled here?

Reply:
This is only for marking fields optional that have been added by this union.
Complex fields are always optional, so they neither set the `unionMember` bool, nor do they need to be iterated here.

Reply:
Resolving as it's no longer used.